src/core/chunked_stream.js

```diff
@@ -30,252 +30,238 @@ var ChunkedStream = (function ChunkedStreamClosure() {
     this.numChunks = Math.ceil(length / chunkSize);
     this.manager = manager;
     this.progressiveDataLength = 0;
-    this.lastSuccessfulEnsureByteChunk = -1;  // a single-entry cache
+    this.lastSuccessfulEnsureByteChunk = -1; // Single-entry cache
   }
 
-  // required methods for a stream. if a particular stream does not
-  // implement these, an error should be thrown
-  ChunkedStream.prototype = {
-
-    getMissingChunks: function ChunkedStream_getMissingChunks() {
-      var chunks = [];
-      for (var chunk = 0, n = this.numChunks; chunk < n; ++chunk) {
-        if (!this.loadedChunks[chunk]) {
-          chunks.push(chunk);
-        }
+  // If a particular stream does not implement one or more of these methods,
+  // an error should be thrown.
+  getMissingChunks() {
+    const chunks = [];
+    for (let chunk = 0, n = this.numChunks; chunk < n; ++chunk) {
+      if (!this.loadedChunks[chunk]) {
+        chunks.push(chunk);
       }
-      return chunks;
-    },
+    }
+    return chunks;
+  }
 
-    getBaseStreams: function ChunkedStream_getBaseStreams() {
-      return [this];
-    },
+  getBaseStreams() {
+    return [this];
+  }
 
-    allChunksLoaded: function ChunkedStream_allChunksLoaded() {
-      return this.numChunksLoaded === this.numChunks;
-    },
+  allChunksLoaded() {
+    return this.numChunksLoaded === this.numChunks;
+  }
 
-    onReceiveData: function ChunkedStream_onReceiveData(begin, chunk) {
-      var end = begin + chunk.byteLength;
+  onReceiveData(begin, chunk) {
+    const chunkSize = this.chunkSize;
+    if (begin % chunkSize !== 0) {
+      throw new Error(`Bad begin offset: ${begin}`);
+    }
+
+    // Using `this.length` is inaccurate here since `this.start` can be moved
+    // (see the `moveStart` method).
+    const end = begin + chunk.byteLength;
+    const length = this.bytes.length;
+    if (end % chunkSize !== 0 && end !== length) {
```

Comment:
Nit: Since `length` appears to be used only once, how do you feel about simply inlining `... && end !== this.bytes.length) {` here instead?